public/app/features/dashboard-scene/settings/version-history/useDashboardRestore.tsx

```diff
@@ -0,0 +1,39 @@
+import { useEffect } from 'react';
+import { useAsyncFn } from 'react-use';
+
+import { locationUtil } from '@grafana/data';
+import { locationService } from '@grafana/runtime';
+import { useAppNotification } from 'app/core/copy/appNotification';
+import { DashboardModel } from 'app/features/dashboard/state';
+import { useSelector } from 'app/types';
+
+import { dashboardWatcher } from '../../../live/dashboard/dashboardWatcher';
+
+import { historySrv } from './HistorySrv';
+
+const restoreDashboard = async (version: number, dashboard: DashboardModel) => {
+  // Skip the watcher logic for this save since it's handled by the hook
+  dashboardWatcher.ignoreNextSave();
+  return await historySrv.restoreDashboard(dashboard, version);
+};
+
+export const useDashboardRestore = (version: number) => {
+  const dashboard = useSelector((state) => state.dashboard.getModel());
+  const [state, onRestoreDashboard] = useAsyncFn(async () => await restoreDashboard(version, dashboard!), []);
+  const notifyApp = useAppNotification();
+
+  useEffect(() => {
+    if (state.value) {
+      const location = locationService.getLocation();
+      const newUrl = locationUtil.stripBaseFromUrl(state.value.url);
+      const prevState = (location.state as any)?.routeReloadCounter;
```

Comment:
Why this is not typed? Is there any tricky one to solve this? 😄

Reply:
It looked a tad tricky yesterday when moving things around, but I didn't spend much on it. I'll give it a another go and see if there's a good fix for this